process/smartContract/process.go

```diff
@@ -834,12 +853,13 @@ func (sc *scProcessor) penalizeUserIfNeeded(
 		return
 	}
 
-	isTooMuchGasProvided := callType != vmcommon.AsynchronousCall && isTooMuchGasProvided(gasProvided, vmOutput.GasRemaining)
+	isTooMuchGasProvided := isTooMuchGasProvided(gasProvided, vmOutput.GasRemaining)
```

Comment:
the check is a must here. as isTooMuchGasProvided has to now about what value was kept back, and not to delete it.

Reply:
Reverted.